DiscoveryEngineTargetSite: initial types

justinsb opened the pull request:
- **DiscoveryEngineTargetSite: initial types**
- **DiscoveryEngineTargetSite: generated apimachinery**

yuwenma commented on apis/discoveryengine/v1alpha1/targetsite_reference.go:
```diff
@@ -0,0 +1,155 @@
+}
+
+// NewTargetSiteLinkFromObject builds a TargetSiteLink from the Config Connector object.
+func NewTargetSiteLinkFromObject(ctx context.Context, reader client.Reader, obj *DiscoveryEngineDataStoreTargetSite) (*DiscoveryEngineDataStoreID, *TargetSiteLink, error) {
```
Just curious, this is not called and also not from the SciFi template. Is it from Gemini?

yuwenma commented on apis/discoveryengine/v1alpha1/targetsite_types.go:
```diff
@@ -0,0 +1,118 @@
+
+	// Required. Input only. The user provided URI pattern from which the
+	// `generated_uri_pattern` is generated.
+	ProvidedUriPattern *string `json:"providedUriPattern,omitempty"`
```
nit: Uri --> URI

yuwenma replied:
Do you mind adding this acronym to the CRD validation list? https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/tests/apichecks/crds_test.go#L195

yuwenma commented on apis/discoveryengine/v1alpha1/targetsite_types.go:
```diff
@@ -0,0 +1,118 @@
+	GeneratedUriPattern *string `json:"generatedUriPattern,omitempty"`
+
+	// Output only. Root domain of the provided_uri.
+	RootDomainUri *string `json:"rootDomainUri,omitempty"`
```
nit: Uri --> URI

yuwenma commented on apis/discoveryengine/v1alpha1/targetsite_types.go:
```diff
@@ -0,0 +1,118 @@
+// +kcc:proto=google.cloud.discoveryengine.v1.TargetSite
+type DiscoveryEngineDataStoreTargetSiteObservedState struct {
+	// Output only. This is system-generated based on the provided_uri.
+	GeneratedUriPattern *string `json:"generatedUriPattern,omitempty"`
```
ditto

yuwenma reviewed (commented):
/approve

justinsb commented:
Added URI to acronym validation and updated the json field tags.  I didn't update the go field names because the mapper uses those and is case sensitive (though I think elsewhere I added "near misses" to deal with this problem)

yuwenma commented:
Looks great. Thank! Waiting for presubmit to pass

yuwenma commented:
/lgtm
